### PR TITLE
View switcher

### PR DESCRIPTION
**Pull request by warrickct**
* Probably best to do this PR second based on commits it contains.
* Contains updated graphql query to include the viewType field.
* Adds dynamic view switching for the subhub pages based on contentful response data.

### CODE REVIEW

**Comment by JuliusKarl** [comment]
- Initial wireframes for ResearchHub Expansion redesign (_Article and Equipment_)
- View Switches (_2 Options_) from Contentful Option
- Additional GraphQL Query implemented to bypass complexity issue (_GetArticleByID_ and _GetEquipmentByID_)

**Review by Hganavak — CHANGES_REQUESTED** [review]
Same as #38,  @warrickct  if you can make the requested changes to #36. We'll get those merged into `master`, then you can just merge `master` into these 2 branches, re-request review and be good to go :v:

File: Jenkinsfile [hunk 1]
```diff
@@ -91,14 +91,50 @@ pipeline {
...
+                        }
+                        stage ('Unzipping existing cached node_modules.') {
+                            when {
+                                not {
+                                    anyOf {
+                                        changeset "**/research-hub-web/package.json"
+                                    }
+                                }
+                            }
+                            steps {
```
Comment (warrickct) [inline Jenkinsfile:122]: These changes had to be done on sandbox in order to test the deployment functionality.

File: research-hub-web/e2e/app.e2e-spec.ts [hunk 2]
```diff
@@ -48,127 +48,80 @@ describe('ResearchHub\'s Basic Functionality', () => {
    * Visits the home page -> Clicks one of the 'Category' tiles -> Checks that the search results page has
    * been successfully navigated to by checking the presence of the search results page title 'Results'.
    */
-  it('can browse by category', async () => {
```
Comment (warrickct) [inline research-hub-web/e2e/app.e2e-spec.ts:51]: Assuming this is us disabling protractor e2e tests for now?

File: research-hub-web/src/app/components/equipment/equipment.component.ts [hunk 3]
```diff
@@ -15,12 +25,17 @@ export class EquipmentComponent implements OnInit {
   public allEquipment$: Observable<EquipmentCollection>;
   public equipment$: Observable<Equipment>;
   public slug: string;
+  // public assets: Array<any>;
```
Comment (warrickct) [inline research-hub-web/src/app/components/equipment/equipment.component.ts:28]: Old code comments probably best to remove.

File: research-hub-web/src/app/components/articles/articles.component.ts [hunk 4]
```diff
@@ -45,7 +51,15 @@ export class ArticlesComponent implements OnInit {
      * therefore run the corresponding query. If not, return all articles.
      */
     if (!!this.slug) {
-      this.article$ = this.getArticleBySlug(this.slug);
+      this.getArticleBySlug(this.slug).subscribe(data => {
+        this.article$ = this.getArticleByID(data.sys.id);
+        // this.article$.subscribe(data => {
```
Comment (warrickct) [inline research-hub-web/src/app/components/articles/articles.component.ts:56]: Code comments to be removed.

File: research-hub-web/src/styles.scss [hunk 5]
```diff
@@ -31,6 +31,7 @@ h3 {
 h4 {
```
Comment (warrickct) [inline research-hub-web/src/styles.scss:31]: In the file above "research-hub-web/src/assets/env.js" if it's empty can this be deleted?

File: research-hub-web/src/app/components/equipment/equipment.component.ts [hunk 6]
```diff
@@ -38,13 +53,19 @@ export class EquipmentComponent implements OnInit {
      * therefore run the corresponding query. If not, return all articles.
      */
     if (!!this.slug) {
-      this.equipment$ = this.getEquipmentBySlug(this.slug);
+      this.getEquipmentBySlug(this.slug).subscribe(data => {
+        this.equipment$ = this.getEquipmentByID(data.sys.id);
+        // this.equipment$.subscribe(data => {
```
Comment (warrickct) [inline research-hub-web/src/app/components/equipment/equipment.component.ts:58]: Old code to delete

File: research-hub-web/src/app/components/articles/articles.component.ts (hunk 4 above)
Comment (warrickct) [inline research-hub-web/src/app/components/articles/articles.component.ts:56]: old code to delete

File: research-hub-web/src/app/components/articles/articles.component.ts [hunk 7]
```diff
@@ -21,12 +22,17 @@ export class ArticlesComponent implements OnInit {
   public allArticles$: Observable<ArticleCollection>;
   public article$: Observable<Article>;
   public slug: string;
+  // public assets: Array<any>;
```
Comment (warrickct) [inline research-hub-web/src/app/components/articles/articles.component.ts:25]: old code to delete

**Review by warrickct — COMMENTED** [review]
Looks fine to me aside from some bits of old code to be tidied up.

**Comment by warrickct** [comment]
I saw a file "research-hub-web/src/assets/env.js" which is supposedly empty. If it's empty can this be deleted?

**Comment by warrickct** [comment]
@JuliusKarl These are the layout issues I was referring to this morning. The issues are mostly improvements for 4k displays.

* Home page icons could be enlarged.
* New banner titles could be enlarged. 
* There seems to be a bug where a banner in image 2 but I haven't been able to recreate it. Sam said it popped up when he navigated between pages somehow.

I'll have another check to see if there's anything else.

image (4)
---
image
---
image (1)
---
image (3)

**Review by warrickct — COMMENTED** [review]
@JuliusKarl Hey, I noticed a couple of things. Not sure if they're specific to my machine. Could you please run a deployment of all projects? It's so I can see that the serverless and cer-graphql subrepos tests are working as well.

* There seems to be a small strip of another image just above the banner on articles pages. Looks fine for equipment though.
Screen Shot 2020-11-11 at 2 54 29 PM

---

* The option tile headers are very small
Screen Shot 2020-11-11 at 2 56 36 PM

---

* I'm signed in but unable to see any content for the top secret article.
Screen Shot 2020-11-11 at 2 57 26 PM

**Comment by JuliusKarl** [comment]
Hopefully Resolved the latest comments

- The strip is from the original photo
- The font-size is relative to browser size now
- The 'top-secret-entry' was missing a _required_ viewType which is why it was blank. Should be visible now

**Comment by warrickct** [comment]
Ahh ok, my mistake about the top-secret and original photo points. All the tests are passing on my side and it looks all good. Feel free to merge 👍👍 .

**Pull request merged**